File: Container-With-Most-Water/Trapping-Rain-Water-II.py
height = [1,1]

def maxArea(height):
    l =0
    r = len(height)-1
    maxs = 0
    while l<r:
        if height[l]>height[r]:
            maxs = max(height[r]*(r-l),maxs)
            r=r-1
        if height[l]<height[r]:
            maxs = max(height[l]*(r-l),maxs)
            l =l+1
        if height[l]==height[r]:
            maxs = max(height[l]*(r-l),maxs)
            l =l+1
            r=r-1
    return maxs
print(maxArea(height))

File: Container-With-Most-Water/Trapping-Rain-Water-II.js
let height = [1,1]

var maxArea = function(height) {
    let l = 0
    let r = height.length-1
    let max = 0
    while(l<r){
        if(height[l]>height[r]){
            max = Math.max(height[r]*(r-l),max)
            r--
        }
        if(height[l]<height[r]){
            max = Math.max(height[l]*(r-l),max)
            l++
        }
        if(height[l]===height[r]){
            max = Math.max(height[r]*(r-l),max)
            l++
            r--
        }
    }
    return max
};

console.log(maxArea(height))